api/views.py

```diff
@@ -80,30 +81,60 @@ def create(self, request, *args, **kwargs):
         return response
 
 
+class AssetFilter(filters.FilterSet):
+    email = filters.CharFilter(method='filter_email')
+    model_number = filters.CharFilter(method='filter_model_number')
+    asset_type = filters.CharFilter(method='filter_asset_type')
+
+    def filter_email(self, queryset, name, value):
+        _filters = {
+            'model_number': self.filter_model_number,
+            'asset_type': self.filter_asset_type
+        }
+        try:
+            """
+                Since the incoming queryset can be for the logged in user,\
+                creating a queryset for this email ensures subsequent queries\
+                are performed against this user.
+            """
+            query_params = self.request.query_params
+
+            validate_email(value)
+            user_asset_assignee = User.objects.get(email=value)
+            queryset = Asset.objects.filter(assigned_to__user=user_asset_assignee)
+            for query, value in query_params.items():
+                if query == 'email':
+                    pass
+                else:
+                    queryset = _filters[query](queryset, name, value)
+        except (ValidationError, ObjectDoesNotExist):
+            queryset = Asset.objects.none()
```

Comment:
Hello @collinmutembei, @egichuri Previously, a `ValidationError` was being raised at this point. Should this still be the case or we can just return an empty queryset? If we raise a  `ValidationError` here, does it mean we should raise the error even when an `asset_type` or `model_number` does not exist instead of returning empty lists??

Reply:
@esirK was the validation error from `validate_email(value)`?

Reply:
Yes it was @egichuri .

Reply:
you could catch the exceptions separately
```
try:
    # as it is
except ValidationError:
    # email validation error
except ObjectDoesNotExist:
    # empty queryset
```